cdap-messaging-spi/src/main/java/io/cdap/cdap/messaging/spi/MessagingService.java

```diff
@@ -39,9 +39,10 @@ public interface MessagingService {
    * Initialize the messaging service. This method is guaranteed to be called before any other
    * method is called. It will only be called once for the lifetime of the messaging service.
    *
-   * @param context the context that can be used to initialize the messaging service.
+   * @param context      the context that can be used to initialize the messaging service.
+   * @param systemTopics topics to be created initially.
    */
-  void initialize(MessagingServiceContext context) throws IOException;
+  void initialize(MessagingServiceContext context, List<TopicMetadata> systemTopics) throws IOException;
```

Comment:
Better to pass systemTopics through MessagingServiceContext as context already has cConf and AFAICT topicsToCreate() just needs cConf. So we just need to add `getSystemTopics` method in context. 
WDYT?

Reply:
Thats a nice suggestion. Updated the code accordingly.
Thanks!